inst/templates/ad_adpet.R

```diff
@@ -0,0 +1,211 @@
+
+# Use e.g. `haven::read_sas()` to read in .sas7bdat, or other suitable functions
+# as needed and assign to the variables below.
+# For illustration purposes read in admiral test data
+nv <- admiralneuro::nv_neuro
+ag <- admiralneuro::ag_neuro
+suppnv <- admiralneuro::suppnv_neuro
+adsl <- admiralneuro::adsl_neuro
+
+# When SAS datasets are imported into R using haven::read_sas(), missing
+# character values from SAS appear as "" characters in R, instead of appearing
+# as NA values. Further details can be obtained via the following link:
+# https://pharmaverse.github.io/admiral/articles/admiral.html#handling-of-missing-values # nolint
+nv <- convert_blanks_to_na(nv)
+
+
+# Combine the parental datasets with their respective supp datasets (only if exist)
+# User can use `combine_supp()` from {metatools} to combine the parental with supp dataset.
+
+nv <- metatools::combine_supp(nv, suppnv)
+
+# Lookup tables ----
+
+# Assign PARAMCD, PARAM, and PARAMN
+param_lookup <- tibble::tribble(
+  ~NVTESTCD, ~NVCAT, ~NVLOC, ~REFREG, ~PARAMCD, ~PARAM, ~PARAMN,
+  "SUVR", "FBP", "NEOCORTICAL COMPOSITE", "Whole Cerebellum", "SNCWCFBP", "FBP Standard Uptake Ratio Neocortical Composite Whole Cerebellum", 1,
+  "SUVR", "FBB", "NEOCORTICAL COMPOSITE", "Whole Cerebellum", "SNCWCFBB", "FBB Standard Uptake Ratio Neocortical Composite Whole Cerebellum", 2,
+  "SUVR", "FTP", "NEOCORTICAL COMPOSITE", "Inferior Cerebellar Gray Matter", "SNCTFTP", "FTP Standard Uptake Ratio Neocortical Composite Inferior Cerebellar Gray Matter", 3,
+  "VR", "FBP", NA, NA, "VRFBP", "FBP Qualitative Visual Classification", 4,
+  "VR", "FTP", NA, NA, "VRFTP", "FTP Qualitative Visual Classification", 5
+)
+attr(param_lookup$NVTESTCD, "label") <- "NV Test Short Name"
+
+# Derivations ----
+
+# Get list of ADSL vars required for derivations
+adsl_vars <- exprs(TRTSDT, TRTEDT, TRT01A, TRT01P)
+
+adpet <- nv %>%
+  ## Join ADSL with NV (need TRTSDT for ADY derivation) ----
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = get_admiral_option("subject_keys")
+  ) %>%
+  ## Join ADPET with AG for tracer information ----
+  # Users can add more variables in the `new_vars` argument as needed.
+  derive_vars_merged(
+    dataset_add = ag,
+    new_vars = exprs(AGTRT, AGCAT),
+    by_vars = exprs(USUBJID, VISIT, NVLNKID = AGLNKID)
+  ) %>%
+  ## Calculate ADT, ADY ----
+  derive_vars_dt(
+    new_vars_prefix = "A",
+    dtc = NVDTC
+  ) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT))
+
+adpet <- adpet %>%
+  ## Add PARAMCD only - add PARAM etc later ----
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup,
+    new_vars = exprs(PARAMCD, PARAM),
+    by_vars = exprs(NVTESTCD, NVCAT, NVLOC)
+  ) %>%
+  ## Calculate AVAL and AVALC ----
+  # AVALC should only be mapped if it contains non-redundant information.
+  mutate(
+    AVAL = NVSTRESN,
+    AVALC = if_else(
+      is.na(NVSTRESN) | as.character(NVSTRESN) != NVSTRESC,
```

Comment:
Was this supposed to be `==` instead of `!=` ?
```  
      is.na(NVSTRESN) | as.character(NVSTRESN) == NVSTRESC,
``` 

Else,  `as.character(NVSTRESN) != NVSTRESC` is not needed.

Reply:
I think this is correct as it is. You only want to populate `AVALC` if it contains information that is different to `AVAL` - this is as per CDISC guidance, which the {admiral} team has implemented in the last year after feedback that we were non-compliant. Consequently you check whether `as.character(NVSTRESN)` matches `NVSTRESC`, and if it does not, then you can populate `AVALC` with this new info, otherwise you leave it.

Reply:
Ah okay, yeah that makes sense. Thanks!